From ab09be7e44169ed8f568d292768cb2492e1b68ec Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:50:00 -0400
Subject: [pci] Don't clear IORESOURCE_UNSET when updating BAR

Message-id: <20140806035000.1438.75533.stgit@gir.stowe>
Patchwork-id: 87464
O-Subject: [RHEL7.1 PATCH 55/69] PCI: Don't clear IORESOURCE_UNSET when updating BAR
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: 434aafc1aefb5eb6e8c8d15284c8f929be756521

commit 434aafc1aefb5eb6e8c8d15284c8f929be756521
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Wed Feb 26 11:25:59 2014 -0700

    PCI: Don't clear IORESOURCE_UNSET when updating BAR

    Clear IORESOURCE_UNSET when we assign an address to a resource, not when we
    write the address to the BAR.

    Also, drop the "BAR %d: set to %pR" message; this is mostly redundant with
    the "BAR %d: assigned %pR" message from pci_assign_resource().

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/setup-res.c b/drivers/pci/setup-res.c
index 0474b02..725d5b2 100644
--- a/drivers/pci/setup-res.c
+++ b/drivers/pci/setup-res.c
@@ -101,11 +101,6 @@ void pci_update_resource(struct pci_dev *dev, int resno)
 
  if (disable)
   pci_write_config_word(dev, PCI_COMMAND, cmd);
-
- res->flags &= ~IORESOURCE_UNSET;
- dev_dbg(&dev->dev, "BAR %d: set to %pR (PCI address [%#llx-%#llx])\n",
-  resno, res, (unsigned long long)region.start,
-  (unsigned long long)region.end);
 }
 
 int pci_claim_resource(struct pci_dev *dev, int resource)
-- 
1.7.1